src/components/remote_control/include/remote_control/resource_allocation_manager_impl.h

```diff
@@ -0,0 +1,79 @@
+#ifndef SRC_COMPONENTS_REMOTE_CONTROL_INCLUDE_REMOTE_CONTROL_RESOURCE_ALLOCATION_IMPL_H
+#define SRC_COMPONENTS_REMOTE_CONTROL_INCLUDE_REMOTE_CONTROL_RESOURCE_ALLOCATION_IMPL_H
+#include "remote_control/resource_allocation_manager.h"
+#include "remote_control/remote_plugin_interface.h"
+#include "utils/macro.h"
+namespace remote_control {
+
+typedef rc_event_engine::EventDispatcher<application_manager::MessagePtr,
+                                         std::string> RCEventDispatcher;
+
+class ResourceAllocationManagerImpl : public ResourceAllocationManager {
+ public:
+  ResourceAllocationManagerImpl(RemotePluginInterface& rc_plugin);
+
+  ~ResourceAllocationManagerImpl();
+
+  AcquireResult::eType AcquireResource(const std::string& module_type,
```

Comment:
please add function descriptions

Reply:
@LuxoftAKutsan Still needs description.

Reply:
@JackLivio descriptions are available in interface class - ResourceAllocationManager